src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -78,6 +79,28 @@ class LockfileContent:
     lockfile_hex_digest: str | None
 
 
+@dataclass(frozen=True)
+class _ToolLockfileMixin:
+    options_scope_name: str
+    uses_source_plugins: bool
+    uses_project_interpreter_constraints: bool
+
+
+@dataclass(frozen=True)
+class ToolDefaultLockfile(LockfileContent, _ToolLockfileMixin):
+    pass
+
+
+@dataclass(frozen=True)
+class ToolCustomLockfile(Lockfile, _ToolLockfileMixin):
+    pass
+
+
+@dataclass(frozen=True)
+class ToolProgrammaticLockfile(LockfileContent, _ToolLockfileMixin):
+    pass
```

Comment:
I don't think this makes sense to have without having call sites using this. 

Instead, I think we simply don't support anything other than `ToolCustomLockfile ` and `ToolDefaultLockfile` for now, as mentioned below.

Reply:
Fair